### PR TITLE
Repository Sets: make name & product optional inplace of label

### PR DESCRIPTION
**Pull request by sean797**
Requires ~~SatelliteQE/nailgun#574~~

### CODE REVIEW

**Comment by mdellweg** [comment]
Another test. Yay!

File: module_utils/ansible_nailgun_cement.py [hunk 1]
```diff
@@ -520,9 +520,12 @@ def find_content_credential(module, name, organization, failsafe=False):
     return handle_find_response(module, content_credential.search(), message="No content credential found for %s" % name, failsafe=failsafe)
 
 
-def find_repository_set(module, name, product, failsafe=False):
-    repo_set = RepositorySet(name=name, product=product)
-    return handle_find_response(module, repo_set.search(), message="No repository set found for %s" % name, failsafe=failsafe)
+def find_repository_set(module, name, product, organization=None, label=None, failsafe=False):
```
Comment (akofink) [inline module_utils/ansible_nailgun_cement.py]: IMO this should be:

```py
def find_repository_set(module, name, product=None, organization=None, failsafe=False)
```

Repo sets index API takes either organization ID or product ID (and repo set name) to resolve the repo set ID. I'm confused what this label is. Care to explain?

Comment (akofink) [reply module_utils/ansible_nailgun_cement.py]: I don't understand. I'm looking at our [API controller params](https://github.com/Katello/katello/blob/master/app/controllers/katello/api/v2/repository_sets_controller.rb#L23) and don't see label. It looks like you can search with either `org ID + name` or `product ID + name`.

Comment (sean797) [reply module_utils/ansible_nailgun_cement.py]: @akofink the label search uses scoped_search, which does allow us to search via label  https://github.com/Katello/katello/blob/master/app/models/katello/product_content.rb#L23

Comment (akofink) [reply module_utils/ansible_nailgun_cement.py]: Ah, ok that makes sense. :)

Comment (akofink) [reply module_utils/ansible_nailgun_cement.py]: So then, why is name required here in the method definition? You can use either name or label, right? So it should look like this:

```py
def find_repository_set(module, name=None, label=None, product=None, organization=None, failsafe=False)
```

This allows using name or label and product or organization. Maybe I'm missing something.

Comment (sean797) [reply module_utils/ansible_nailgun_cement.py]: Yup, nice spot! Updated :+1: Thanks!

**Comment by sean797** [comment]
Updated and rebased :green_apple: - thanks @akofink

File: module_utils/ansible_nailgun_cement.py [hunk 2]
```diff
@@ -520,9 +520,12 @@ def find_content_credential(module, name, organization, failsafe=False):
...
 
 
-def find_repository_set(module, name, product, failsafe=False):
-    repo_set = RepositorySet(name=name, product=product)
-    return handle_find_response(module, repo_set.search(), message="No repository set found for %s" % name, failsafe=failsafe)
+def find_repository_set(module, name=None, product=None, organization=None, label=None, failsafe=False):
+    if label and organization:
+        repo_set = RepositorySet(organization=organization).search(query={'search': 'label="{}"'.format(label)})
+    else:
+        repo_set = RepositorySet(name=name, product=product).search()
```
Comment (akofink) [inline module_utils/ansible_nailgun_cement.py]: The error is here. With the input I provided (name and org only), `product` is `None`.

**Comment by sean797** [comment]
I don't think that's ever worked, I'm surprised to hear it works with hammer. If we just supply `name` and `org` how do we know what product to use?

File: modules/katello_repository_set.py [hunk 3]
```diff
@@ -132,11 +151,12 @@ def get_desired_repos(desired_substitutions, available_repos):
     return desired_repos
 
 
-def repository_set(module, name, organization, product, state, repositories=[]):
+def repository_set(module, name, organization, product, label, state, repositories=[]):
     changed = False
     organization = find_organization(module, organization)
-    product = find_product(module, product, organization)
```
Comment (sean797) [inline modules/katello_repository_set.py]: Looking at this old line, I'm fairly sure that never worked.. I'd vote for merging this as is.. If we want we can support that feature in the future :-)

**Comment by akofink** [comment]
I actually think this is a fairly common use case. Repository set names (and labels) should be unique within an organization. We're really close to feature parity here with the API/CLI; I'd push to just tweak it a bit more; however, if you insist, we can merge as is and file a new bug to allow name+org only.

**Comment by sean797** [comment]
Didn't realise it was going to be that simple - Updated and I added a test for name + org ;)

**Pull request merged**